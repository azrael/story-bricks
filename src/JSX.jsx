import React, { Children } from 'react';
import SyntaxHighlighter from 'react-syntax-highlighter/dist/cjs/prism-light';
import jsx from 'react-syntax-highlighter/dist/cjs/languages/prism/jsx';
import xonokai from 'react-syntax-highlighter/dist/cjs/styles/prism/xonokai';
import reactElementToJSXString from 'react-element-to-jsx-string';

SyntaxHighlighter.registerLanguage('jsx', jsx);

const options = {
    showDefaultProps: false,
    tabStop: 4
};

const JSX = ({ children }) => <SyntaxHighlighter language='jsx' style={xonokai}>
    {Children
        .toArray(children)
        .map(child => typeof child === 'string' ? child : reactElementToJSXString(child, options))
        .join('\n')
    }
</SyntaxHighlighter>;

JSX.displayName = 'JSX';

export default JSX;
