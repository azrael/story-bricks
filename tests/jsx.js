import React from 'react';
import renderer from 'react-test-renderer';
import JSX from '../src/JSX.jsx';

it('JSX with single child renders correctly', () => {
    const tree = renderer
        .create(<JSX><div>test</div></JSX>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it('JSX with multiple children renders correctly', () => {
    const tree = renderer
        .create(<JSX>
            <div>foo</div>
            bar
            <span>baz</span>
        </JSX>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it('JSX with no children renders correctly', () => {
    const tree = renderer
        .create(<JSX />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});
