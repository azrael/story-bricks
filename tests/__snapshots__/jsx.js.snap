// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSX with multiple children renders correctly 1`] = `
<pre
  style={
    Object {
      "MozHyphens": "none",
      "MozTabSize": "2",
      "OTabSize": "2",
      "WebkitHyphens": "none",
      "background": "#2a2a2a",
      "border": "1px solid #e1e1e8",
      "borderRadius": "4px",
      "color": "#76d9e6",
      "fontFamily": "Menlo, Monaco, \\"Courier New\\", monospace",
      "fontSize": "14px",
      "hyphens": "none",
      "msHyphens": "none",
      "overflow": "auto",
      "padding": "15px",
      "position": "relative",
      "tabSize": "2",
      "textShadow": "none",
      "whiteSpace": "pre-wrap",
      "wordWrap": "normal",
    }
  }
>
  <code
    style={
      Object {
        "MozHyphens": "none",
        "MozTabSize": "2",
        "OTabSize": "2",
        "WebkitHyphens": "none",
        "color": "#76d9e6",
        "fontFamily": "Menlo, Monaco, \\"Courier New\\", monospace",
        "fontSize": "14px",
        "hyphens": "none",
        "msHyphens": "none",
        "tabSize": "2",
        "textShadow": "none",
        "whiteSpace": "pre-wrap",
        "wordWrap": "normal",
      }
    }
  >
    <span
      className="token tag"
      style={Object {}}
    >
      <span
        className="token tag"
        style={Object {}}
      >
        <span
          className="token"
          style={
            Object {
              "color": "#bebec5",
            }
          }
        >
          &lt;
        </span>
        div
      </span>
      <span
        className="token"
        style={
          Object {
            "color": "#bebec5",
          }
        }
      >
        &gt;
      </span>
    </span>
    <span
      className="token plain-text"
      style={Object {}}
    >
      
    foo

    </span>
    <span
      className="token tag"
      style={Object {}}
    >
      <span
        className="token tag"
        style={Object {}}
      >
        <span
          className="token"
          style={
            Object {
              "color": "#bebec5",
            }
          }
        >
          &lt;/
        </span>
        div
      </span>
      <span
        className="token"
        style={
          Object {
            "color": "#bebec5",
          }
        }
      >
        &gt;
      </span>
    </span>
    
bar

    <span
      className="token tag"
      style={Object {}}
    >
      <span
        className="token tag"
        style={Object {}}
      >
        <span
          className="token"
          style={
            Object {
              "color": "#bebec5",
            }
          }
        >
          &lt;
        </span>
        span
      </span>
      <span
        className="token"
        style={
          Object {
            "color": "#bebec5",
          }
        }
      >
        &gt;
      </span>
    </span>
    <span
      className="token plain-text"
      style={Object {}}
    >
      
    baz

    </span>
    <span
      className="token tag"
      style={Object {}}
    >
      <span
        className="token tag"
        style={Object {}}
      >
        <span
          className="token"
          style={
            Object {
              "color": "#bebec5",
            }
          }
        >
          &lt;/
        </span>
        span
      </span>
      <span
        className="token"
        style={
          Object {
            "color": "#bebec5",
          }
        }
      >
        &gt;
      </span>
    </span>
  </code>
</pre>
`;

exports[`JSX with no children renders correctly 1`] = `
<pre
  style={
    Object {
      "MozHyphens": "none",
      "MozTabSize": "2",
      "OTabSize": "2",
      "WebkitHyphens": "none",
      "background": "#2a2a2a",
      "border": "1px solid #e1e1e8",
      "borderRadius": "4px",
      "color": "#76d9e6",
      "fontFamily": "Menlo, Monaco, \\"Courier New\\", monospace",
      "fontSize": "14px",
      "hyphens": "none",
      "msHyphens": "none",
      "overflow": "auto",
      "padding": "15px",
      "position": "relative",
      "tabSize": "2",
      "textShadow": "none",
      "whiteSpace": "pre-wrap",
      "wordWrap": "normal",
    }
  }
>
  <code
    style={
      Object {
        "MozHyphens": "none",
        "MozTabSize": "2",
        "OTabSize": "2",
        "WebkitHyphens": "none",
        "color": "#76d9e6",
        "fontFamily": "Menlo, Monaco, \\"Courier New\\", monospace",
        "fontSize": "14px",
        "hyphens": "none",
        "msHyphens": "none",
        "tabSize": "2",
        "textShadow": "none",
        "whiteSpace": "pre-wrap",
        "wordWrap": "normal",
      }
    }
  />
</pre>
`;

exports[`JSX with single child renders correctly 1`] = `
<pre
  style={
    Object {
      "MozHyphens": "none",
      "MozTabSize": "2",
      "OTabSize": "2",
      "WebkitHyphens": "none",
      "background": "#2a2a2a",
      "border": "1px solid #e1e1e8",
      "borderRadius": "4px",
      "color": "#76d9e6",
      "fontFamily": "Menlo, Monaco, \\"Courier New\\", monospace",
      "fontSize": "14px",
      "hyphens": "none",
      "msHyphens": "none",
      "overflow": "auto",
      "padding": "15px",
      "position": "relative",
      "tabSize": "2",
      "textShadow": "none",
      "whiteSpace": "pre-wrap",
      "wordWrap": "normal",
    }
  }
>
  <code
    style={
      Object {
        "MozHyphens": "none",
        "MozTabSize": "2",
        "OTabSize": "2",
        "WebkitHyphens": "none",
        "color": "#76d9e6",
        "fontFamily": "Menlo, Monaco, \\"Courier New\\", monospace",
        "fontSize": "14px",
        "hyphens": "none",
        "msHyphens": "none",
        "tabSize": "2",
        "textShadow": "none",
        "whiteSpace": "pre-wrap",
        "wordWrap": "normal",
      }
    }
  >
    <span
      className="token tag"
      style={Object {}}
    >
      <span
        className="token tag"
        style={Object {}}
      >
        <span
          className="token"
          style={
            Object {
              "color": "#bebec5",
            }
          }
        >
          &lt;
        </span>
        div
      </span>
      <span
        className="token"
        style={
          Object {
            "color": "#bebec5",
          }
        }
      >
        &gt;
      </span>
    </span>
    <span
      className="token plain-text"
      style={Object {}}
    >
      
    test

    </span>
    <span
      className="token tag"
      style={Object {}}
    >
      <span
        className="token tag"
        style={Object {}}
      >
        <span
          className="token"
          style={
            Object {
              "color": "#bebec5",
            }
          }
        >
          &lt;/
        </span>
        div
      </span>
      <span
        className="token"
        style={
          Object {
            "color": "#bebec5",
          }
        }
      >
        &gt;
      </span>
    </span>
  </code>
</pre>
`;
